utils/lib/Component.ts

```diff
@@ -0,0 +1,51 @@
+import * as path from 'path';
+import * as fs from 'fs';
+import * as yaml from 'js-yaml';
+
+abstract class Component<ConfigType, MutationVariablesType> {
+  public localPath: string; // Local path to the component. Ex: python/flask
+  public configPath: string; // Absolute path to the config file within the repository
+  public config: ConfigType;
+  public isValid = true;
+  public basePath: string;
+
+  constructor(
+    localPath: string,
+    basePath: string = path.join(__dirname, '../..')
```

Comment:
If we _wanted_ to, we could use a shorthand here: https://dev.to/satansdeer/typescript-constructor-shorthand-3ibd